### ISSUE TITLE
Stream is barely visible on scrub

### ISSUE THREAD

**Issue by sommerluk** [gravitystorm/openstreetmap-carto#2098]
Stream is barely visible on scrub.

The waterway=stream is barely visible on the natural=scrub, especially when combined with intermittent=yes. Screenshot from http://www.openstreetmap.org/#map=19/5.34678/-3.98195
![screenshot](https://cloud.githubusercontent.com/assets/6830724/14065014/d3881fe2-f407-11e5-8386-c8bb1582355a.png)

Not sure what would be a good solution for this, if any…

**Comment by SomeoneElseOSM** [comment]
It's bug-for-bug compatible with real life - even on the ground in summer you sometimes can't tell where the streams would run after it's been raining :)

**Comment by moliha** [comment]
But the map should help finding streams and other waterways. This could be easier, if the color used for waterways had better contrast to the various greens used for forest, scrub etc. I think a darker and more blue color for waterways would be better.

**Comment by kocio-pl** [comment]
Similar to https://github.com/gravitystorm/openstreetmap-carto/issues/1781.

**Comment by kocio-pl** [comment]
Unfortunately still the same with a new water color.

**Comment by HolgerJeromin** [comment]
for reference 
![image](https://user-images.githubusercontent.com/2410353/30629847-4a19f9ba-9dde-11e7-989e-7edfd4685e9a.png)

**Comment by Adamant36** [comment]
@kocio-pl making scrub a tad darker seems to help if that's an option. I think its a tad to light. 
![scrub with stream alternate](https://user-images.githubusercontent.com/30259065/44945466-f4e6bb00-ad9e-11e8-89bb-7dfc7670470f.png)
Its just a mock up, but it looks better. Maybe it could be a little lighter if need be and still have the same effect.

**Comment by kocio-pl** [comment]
All the colors look different, did you use Chrome for a screenshot? It has known problems with color profile.

**Comment by Adamant36** [comment]
I just did it in paint with the paint can. The only things I changed where were the scrub fill color and some of the plant icons. I'll have to test it in kosmtik though now that you mention it.

**Comment by Tomasz-W** [comment]
@Adamant36 

Here is a visualisation of different green shades used for certain plants in _osm-carto_.

![zielen](https://user-images.githubusercontent.com/25656654/44992219-eb27a800-af96-11e8-8d55-81be343a7565.jpg)

As you can see, scrub colour is quite not maching the rest. The fourth square in 50-200 cm column is a scrub pattern on a #aedfa3 (orchard/ vineyard colour), Can you make some test renderings of streams with it?

**Comment by Tomasz-W** [comment]
@Adamant36 Can you take care of this issue next? I'm planning a quite big proposal of coordinating non-landcover green shades (i.a. @_park_  and @_leisure_ change), but of course it has to match another green shades and make complex system, so this issue and https://github.com/gravitystorm/openstreetmap-carto/issues/3411 have to be done first.

Here are my Ps mock-ups with proposed above #AEDFA3. Visibility of a intermittent stream seems to be better. It's not perfect, but we propably won't find better shade without making mess with another green shades.

_(click to view full size!)_
![1](https://user-images.githubusercontent.com/25656654/46128368-811bbf00-c233-11e8-816f-1d1157c4dccd.jpg)
![2](https://user-images.githubusercontent.com/25656654/46128369-811bbf00-c233-11e8-95dc-c8ccb664a60a.jpg)
![3](https://user-images.githubusercontent.com/25656654/46128370-811bbf00-c233-11e8-8676-beb20ed7c23f.jpg)
![4](https://user-images.githubusercontent.com/25656654/46128371-81b45580-c233-11e8-9e9b-d157c865e678.jpg)

**Comment by Adamant36** [comment]
@Tomasz-W, sure. I'll try to get to it in the next couple of days.

**Comment by Tomasz-W** [comment]
@Adamant36 The idea is to just copy "@_orchard_" colour for scrub in landcover code. If we start to play with this colour, we will have to test also orchard, vineyard and plant_nursery compared to grass, forest, etc. I think it's not worth it and IMO it would propably end as a waste of time. Examples above shows nice improvement with using orchard colour and I'm OK with it.

**Comment by Adamant36** [comment]
@Tomasz-W, is there a reason scrub couldn't be changed to more of a golden brown/greyish color since scrub land isn't really green most of the time anyway and there can be a lot of patches of dirt in between the woody plants? To me, a solid green seems to much like woods or that there is consistent ground cover when there isn't. 

Id also suggest the same for grassland/meadow. More of golden brown color. Especially if the color of parks/pitches/sports centers is going to be changed. We could create a scrub/grassland/meadow color category made up of gold/brown and free up green for parks/woods. Otherwise, I don't think any of them are doable or that just adding a pattern to meadows would be adequate. Plus, it just looks more realistic.

**Comment by Tomasz-W** [comment]
@Adamant36 My assumpion is that we have definetely too much unnecessary green shades now, which makes map cluttered and not intuitive  in some areas. Another problem is a threat of close similarity of some colours of different features.

I'm sure that meadows and grasslands (https://github.com/gravitystorm/openstreetmap-carto/issues/3143) are rather green, so we should stay with planned 'grass + pattern' rendering for them. Heath (https://github.com/gravitystorm/openstreetmap-carto/issues/780) has another colour, but very similar height (see the diagram https://github.com/gravitystorm/openstreetmap-carto/issues/780#issuecomment-418130803), so I would include it also in this rendering. As scrubs can be yellow/ brown as well as green, and they have different height, we can try some olive shade for them (maybe heath colour re-use after moving it to green?), but you have to remember about another similars colours in use (see https://wiki.openstreetmap.org/wiki/AreasTab) to not make any conflict. I consider previously proposed solution (https://github.com/gravitystorm/openstreetmap-carto/issues/2098#issuecomment-424980808) as a good enough one, but it's always good to try other options if someone wants to.

**Comment by Adamant36** [comment]
@Tomasz-W, I think grassland is only green for the two or three months a year that it gets rain in most places. Then its more of a high grass, dry golden brown hay color. Which is one of the reasons its used for grassing land and why I proposed it be colored similar to farmland. Meadows on the other hand might be green more often, but they are also usually a mix of grass and wild flowers during "normal" times. So, make sense to not them color meadows as straight green.

**Comment by Tomasz-W** [comment]
@Adamant36 That's why I propose to use pattern, which would suggest that is not a typically green area without making conflicts with other colours, see https://github.com/gravitystorm/openstreetmap-carto/issues/3143#issuecomment-439680472

**Comment by Adamant36** [comment]
Id also add that through a lot of research I have done, at least in america there is a major amount of miss tagging when it comes to grassland/meadows/grass/heath/scrub. Heath seems to be a mainly European thing, although there is a lot of things here miss tagged as it that should probably be scrub (or maybe visa versa?). Whereas, know one seems to know the difference between a meadow, grassland, or grass. While I think the issue is partly due to them being landuse instead of landcover tags, the very similar similar colors also probably don't help (1). Giving grassland a pattern fill would help slightly, but I think giving them drastically, more realistic, colors would also. 

1. People in America also seem to miss tag whole parks as playgrounds a lot to. That could be further evidence for the fact that coloring different things very similar leads to tagging problems or it could just allude to the fact that Americans as a group are extremely bad at the subtleties involved in things like this. Who knows. 

P.s. The pattern might work for meadows, but I don't think it does for grassland. As grassland isn't green most of the time in most places and its not a "patterned" looking thing in sky shots either. Also, it doesn't help free up green colors or deal with the issues mentioned above this paragraph of miss tagging. Which I think making grassland part of farmland (most of the time its managed for hay) would.

**Comment by jeisenbe** [comment]
Now that natural areas and landcover are being rendered at z7, z6 and z7,
we should consider how a new scrub color will work when mixed with other
landcovers.

I would like to suggest simplifying the number of different natural colors
used at low zoom, just like we render all developed land the same at low
zoom.

For example, scrub, orchards and swamp can all be the same shade of darker
green, and grass, heath, meadow etc can be one shade of lighter green.

To make this work best, it would be nice if the color for scrub is between
forest and grass, but darker than heath, so there is a gradation: woodland
-> scrubland -> heath -> grass.

I think this will be more understandable for map users; darker is higher
vegetation.

**Comment by Adamant36** [comment]
> just like we render all developed land the same at low
zoom
There's actually an issue about that with quarries being rendered the same as residential areas when zoomed out #2905. So, that type of unification isn't always necessarily a good thing. Also, what about with more zoomed in levels? I agree with you that there should be a gradation, but I'm not sure how that would work with leisure stuff also being green/darker green. 

There's a big need to free up green stuff in general. You might be correct about grassland. It makes sense. I've looked around California and done searchers on Google images for references. I still think though that making it more hay color or similar to farmland would be a good idea. Even if it might not as realistic. Especially since farmland is being rendered at low zooms levels now and it would be cool if it was chunked in with it on that since its a part of farming infrastructure. 

Scrub, I don't know. It could be colored something else, but I guess green would work also. @Tomasz-W, can you come up with a color block thing with the plant colors you already did in the comments above and add in the different leisure colors, along with your new proposed park color? We might need a natural green color meta ticket or a natural/leisure green color meta ticket. I'm not sure we can change them in isolation of each other. 

@jeisenbe, say green is over used and we need to free it up some. What are you willing to sacrifice if not scrub or grassland?

**Comment by jeisenbe** [comment]
Pitches don’t have to be green; they can be any surface; concrete, sand,
grass, clay. Same with sports centers and other leisure, though green is
probably reasonable.

I’d say green should first be used for natural landcover, with woodland as
1st priority.

I do support rendering meadow the same as farmland at low zoom, because it
is also a type of agricultural landuse.

At high zoom levels we could try making meadow a little more similar to
farmland, but it will be hard to change it without making them too similar.

**Comment by Tomasz-W** [comment]
> We might need a natural green color meta ticket or a natural/leisure green color meta ticket. I'm not sure we can change them in isolation of each other.

@Adamant36 They are listed at the end of https://github.com/gravitystorm/openstreetmap-carto/issues/3298#issuecomment-405039682. I've opened https://github.com/gravitystorm/openstreetmap-carto/issues/3517 because they are connected to each other by "lighten/ darken" and they all have simillar functions. I think that in physical green areas case meta-ticket won't help because anyway we have to test different rendering options for each one, one by one. 

If you or @jeisenbe or anybody else wants to work on this, at this stage I think that the best way to solve this mess would be to simply put test renderings of landcover-mixed areas in certain tickets. We would have a comparsion to another green areas then, and if it will be needed, we always can reference another tickets.

**Comment by jeisenbe** [comment]
I've been designing a more print-friendly version of this style, and one of the changes I have tried was changing scrub and heath color to look more harmonious with forest and grassland, so that there is a clear progression from low/annual vegetation to high/perennial vegetation. 

I'm using @grass: #cdebb0; (same), @heath: #cae2a5; @scrub: #badba4; @forest: #add19e; (same)

Here is an examples, next to farmland, untagged areas, and residential 
(The scrub is not perfect yet, because I haven't changed the color of the scrub pattern .png to match!)

Current rendering
![scrub-heath-current](https://user-images.githubusercontent.com/42757252/48673296-699dea00-eb83-11e8-89ae-b1d673b29086.png)

After changing heath to #cae2a5 and scrub: #badba4
![scrub-heath-color-changes](https://user-images.githubusercontent.com/42757252/48673335-80444100-eb83-11e8-91ae-99aa70005abe.png)

**Comment by Tomasz-W** [comment]
For me it's not olive, but just some darker green. I'll suggest some possible colour code later.

**Comment by Tomasz-W** [comment]
My proposition:
* meadow -> @_grass_ + 10% meadow pattern (https://gist.github.com/Tomasz-W/dc2b8fe24c9f50f60b49a40765835d5c)
* heath, grassland-> @_farmland_ or `#DEE5B8` + 10% meadow pattern 
* scrub -> `#DEE5B8` + scrub pattern 

Photoshop mock-ups (sorry for green scrub pattern, I can't leap over it):
_(click to view full size!)_
![meadow3](https://user-images.githubusercontent.com/25656654/48677135-b3c9a000-eb70-11e8-98be-19b99486cdc4.jpg)
![meadow1](https://user-images.githubusercontent.com/25656654/48677109-59304400-eb70-11e8-9b72-9ae7f58f85e0.jpg)
![meadow2](https://user-images.githubusercontent.com/25656654/48677110-59304400-eb70-11e8-988a-fbd6f57cb6d1.jpg)

**Comment by Tomasz-W** [comment]
Another possible one: `#D1E0B4`. I like it even more.

![scrub1](https://user-images.githubusercontent.com/25656654/48677441-600d8580-eb75-11e8-8662-d775a8d44817.jpg)
![scrub2](https://user-images.githubusercontent.com/25656654/48677442-600d8580-eb75-11e8-9891-58a1fd6d94b9.jpg)
![scrub3](https://user-images.githubusercontent.com/25656654/48677443-60a61c00-eb75-11e8-90c1-f40830a41505.jpg)

**Comment by jeisenbe** [comment]
On searching the old threads I found this nice color picker: http://davidjohnstone.net/pages/lch-lab-colour-gradient-picker

Here is the current rendering again, with more intermittent streams added, and the new version with just 2 changes:  heath to #cae2a5 and scrub: #badba4 (the prior images also had other changes from that "printable" branch)

Current with streams:
![scrubheath-intermittent-master](https://user-images.githubusercontent.com/42757252/48679968-9a107300-ebd9-11e8-8cec-9b9de0f3c1f9.png)

Intermittent streams with heath #cae2a5 and scrub #badba4
![scrub-heath-intermittent](https://user-images.githubusercontent.com/42757252/48679970-9e3c9080-ebd9-11e8-8b05-26bd6c1c7b3b.png)

**Comment by jeisenbe** [comment]
Test image with the color for scrub suggested by @Tomasz-W, (but no heath pattern yet)
#d1e0b4
![scrub-same-heath](https://user-images.githubusercontent.com/42757252/48680687-de534180-ebe0-11e8-809b-9fcc946ea529.png)

It might be best to shift the hue of grass/meadow/grassland to be slightly more yellow-green, less saturated, and lighter, because it is quite far from the new farmland color. This will give us more room to work with, in the color space between grass and forest. Farmland is now #eef0d5; Lch(94,14,112). 

Here I am trying #dcecb3 grass/meadow = lch(91,30,119) with d1e0b4for scrub/heath =lch(87, 24, 122) 
![grass-dcecb3-scrubheath-d1e0b4](https://user-images.githubusercontent.com/42757252/48680977-dd231400-ebe2-11e8-8553-6328c9ee3911.png)
Now this looks a little like the Humanitarian style (HDM-Carto).

This would need extensive testing with different environments, because grass in particular can be found in all sorts of places: college campuses, parks, residential areas, etc; and with vineyards, orchards, swamps, marshlands, etc.

I would like to use the scrub color as a background for mangroves, so that's another thing to consider. Any improvement that makes intermittent streams visible should also help with that goal, because the wetland pattern would also be in water color. 

The scrub color change alone would also need extensive testing. 

Would @Adamant36 be able to use the  pattern generator that @imagico made, to try some new png scrub patterns to match the new colors: http://blog.imagico.de/new-pattern-generator-version/

**Comment by jeisenbe** [comment]
Here's another test image with #dcecb3 grass/meadow and d1e0b4for scrub/heath, with a golf area and campsite added for comparison:
![newscrub-grass-withgolf-camp](https://user-images.githubusercontent.com/42757252/48681725-81a75500-ebe7-11e8-9b88-d5074226a9ed.png)

Another area, with a campsite and leisure (festival grounds) area:
Current master:
![lembah-baliem-before](https://user-images.githubusercontent.com/42757252/48681976-502f8900-ebe9-11e8-98b1-a5b87e9da8e2.png)

dcecb3 grass/meadow and d1e0b4for scrub/heath:
![lembah-baliem-after](https://user-images.githubusercontent.com/42757252/48681977-50c81f80-ebe9-11e8-8c54-ce1537f84f25.png)

http://davidjohnstone.net/pages/lch-lab-colour-gradient-picker
Colors at the link are farmland, new grass/meadow, new scrub/heath, forest/wood, orchard/vineyard, park, in that order.

Here's the current colors :http://davidjohnstone.net/pages/lch-lab-colour-gradient-picker#eff0d5,cdebb0,b5e3b5,add19e,aedea3,c8facc,d6d99f
farmland, current grass/meadow, current scrub, forest/wood, orchard/vineyard, park, plus current heath.

**Comment by Adamant36** [comment]
Nice. I really like @jeisenbe's meadow color with @Tomasz-W's scrub color. That was totally what I was thinking about with scrub.

**Comment by jragusa** [comment]
I remind you that the aim of this issue is to improve the readability of intermittent water paths on scrub and not to harmonise all natural landcovers.

**Comment by Adamant36** [comment]
@jragusa, at least "harmonizing all natural landcovers" would help make intermittent water paths on scrubs more visible. At least in theory. Also, its extremely hard to change one landuse color in isolation without considering the colors of other ones along with it.

**Comment by jragusa** [comment]
I agree but in this case it would be better to see intermittent waters on the modified colour to avoid any regressions ;)

**Comment by Adamant36** [comment]
@jragusa, first picture in https://github.com/gravitystorm/openstreetmap-carto/issues/2098#issuecomment-439749088. I'm sure there will be more to.

**Comment by Tomasz-W** [comment]
@jeisenbe 
* at this stage I wouldn't touch grass colour (it's not necessary and we have much other things to do, but we can try with moving similar campsite colour fo it later, colour proposed by you is too yellowish imo)
* using the same colour for heath and scrub (with different patterns) is ok for me.
* I think that golf colour wouldn't fit any natural landcover, but we can consider moving it for pitches, as you wanted to change their colour

**Comment by jeisenbe** [comment]
Here's a new comparison of the current vegetation and green colors:
http://davidjohnstone.net/pages/lch-lab-colour-gradient-picker#eff0d5,cdebb0,d6d99f,b5e3b5,add19e,aedea3,c8facc,aae0cb,b5e3b5,def6c0

The list order is: farmland, grass/meadow, heath, scrub, forest/wood, orchard/vineyard, park, pitch, golf, campsite.

Here is just the scrub change, with heath, grass, and the rest left the same:
http://davidjohnstone.net/pages/lch-lab-colour-gradient-picker#eff0d5,cfecb1,d5d89f,d1e0b4,add19e,aedea3,c8facc,aae0cb,b6e3b5,def6c0

Rendering without the patterns, eg what we would see at lower zoom levels, only other change is d1e0b4 for scrub:
![scrub-d1e0b4-test1](https://user-images.githubusercontent.com/42757252/48746145-70c31600-ecb1-11e8-8472-f5f81ec3992b.png)
![scrub-d1e0b4-test2](https://user-images.githubusercontent.com/42757252/48746146-728cd980-ecb1-11e8-81d9-029238de1def.png)

The new d1e0b4 scrub color has a delta (difference) of 8.4 with the current heath color, and 11.3 with the current wood/forest color. This looks pretty good; its different enough from heath to be distinct, but still somewhat similar. It has a difference of 9.0 with the current campsite color, which is a little close but ok, and 12.5 with farmland, according to the linked color picker.

So we could keep the current heath color and grass color for now, and just change scrub, though it would be nice to have a slightly less brown color for heath.

Re: test renderings with new patterns; I have not yet installed the pattern generator yet.

**Comment by Adamant36** [comment]
@jeisenbe, normally I wouldn't mind waiting to do a PR until we get the other things besides scrub dealt with, but this issue was specific to scrub and the new release is coming out on the 23rd. So it would be good if scrub was included in it in order to see it in the wild and also to honor the intent of the original issue creator. Otherwise, we will have to wait another month until the next version comes out for it to be widely released. There's no reason we can't do that and then work on the other things in a meta ticket or here in the mean time.

**Comment by Adamant36** [comment]
@jeisenbe, can you just do some test renderings with only the scrub color changed that involves intermittent streams please? Its probably hard for people to keep up with the 6 different color changes at once and not get off the topic of how intermittent streams look in the process. Plus, its what the issue is about and its kind of pointless to change the scrub color if it doesn't look good with streams still. Also, the other stuff can be changed later in a meta issue if need be. 

Its better to tackle one thing at a time if its possible. We have already established that the new scrub color looks good, not that it works good with intermittent streams though. I think you should just focus on that, give other people a chance to provide feedback on it, and then move on to the other stuff once we know its good.

**Comment by jeisenbe** [comment]
I've already shown this image with intermittent streams on scrub with the new color suggested by ATomasz-W: d1e0b4
![](https://user-images.githubusercontent.com/42757252/48746145-70c31600-ecb1-11e8-8472-f5f81ec3992b.png)

And this image, with the old png still used for the scrub pattern (this needs to be updated):
![](https://user-images.githubusercontent.com/42757252/48680687-de534180-ebe0-11e8-809b-9fcc946ea529.png)

Here are some real-world examples from Hawaii, with a golf course next door, by chance:
http://www.openstreetmap.org/#map=15/20.7490/-156.8960
_(Should I admit that I just realized that the current scrub background color is the same as the current golf course fill color, b5e3b5? This suggests it would not be perfect as a park color; intermittent stream are rare in developed parks, but not unheard-of)_
z15 current
![manele-before-z15](https://user-images.githubusercontent.com/42757252/48774572-fa083600-ed0d-11e8-8807-5e6bc6188ab5.png)
z16 current
![manele-z16-before](https://user-images.githubusercontent.com/42757252/48774678-40f62b80-ed0e-11e8-99db-d8acbaea35df.png)
z17 current
![manele-z17-before](https://user-images.githubusercontent.com/42757252/48774684-4489b280-ed0e-11e8-888f-375a0b46c0bc.png)

With d1e0b4 for scrub (no pattern yet)
z15 scrub d1e0b4
![manele-z15-scrubonly](https://user-images.githubusercontent.com/42757252/48774708-510e0b00-ed0e-11e8-9179-1d0d4a376b0b.png)
z16 d1e0b4
![manale-z16-scrubonly](https://user-images.githubusercontent.com/42757252/48774718-553a2880-ed0e-11e8-8091-141528e1af1a.png)
z17 d1e0b4
![manele-z17-scrubonly](https://user-images.githubusercontent.com/42757252/48774721-5703ec00-ed0e-11e8-88f9-17a112015eae.png)

**Comment by Tomasz-W** [comment]
@jeisenbe Please add some test renderings with `#d1e0b4` with scrub pattern.

**Comment by jeisenbe** [comment]
More widely spaced pattern:
https://github.com/jeisenbe/openstreetmap-carto/blob/scrub-only/symbols/scrub3.png

![scrub-wider](https://user-images.githubusercontent.com/42757252/48813321-ad5a4480-ed79-11e8-8847-5a10aa8d0781.png)

Hawaii z15
http://www.openstreetmap.org/#map=15/20.7490/-156.8960
![z15-scrub2](https://user-images.githubusercontent.com/42757252/48813322-ae8b7180-ed79-11e8-8854-b3ea8e37ceae.png)

z16
![z16-scrub2](https://user-images.githubusercontent.com/42757252/48813326-b519e900-ed79-11e8-9fb1-1911ef8203d3.png)

**Comment by jeisenbe** [comment]
Maybe the pattern needs to be lighter; the forest/wood pattern is much closer to the background color, and it looks like the current scrub pattern is also closer to the background color than what I've just shown.

Scrub pattern 4: b5c498 pattern on d1e0b4 background
![scrub3-b5c498-d1e0b4-screenshot](https://user-images.githubusercontent.com/42757252/48813785-a03e5500-ed7b-11e8-9aec-14c404ca7593.png)

Test area with scrub 4:
![scrub4-b5c498-d1e0b4-test](https://user-images.githubusercontent.com/42757252/48813875-06c37300-ed7c-11e8-815b-ab582e116e1d.png)

Hawaii z15 with scrub 4:
![scrub4-hawaii-z16](https://user-images.githubusercontent.com/42757252/48814037-af71d280-ed7c-11e8-8871-d6e03434037d.png)

PNG for testing:
![scrub3-b5c498-d1e0b4](https://user-images.githubusercontent.com/42757252/48814065-c6b0c000-ed7c-11e8-9426-b63e579c3107.png)

**Comment by jeisenbe** [comment]
We've been discussing the scrub color change on another issue, https://github.com/gravitystorm/openstreetmap-carto/issues/780 "Improve Natural=Heath Color", and there is a suggestion to use a 3% darker color than the #d1e0b4 that has been shown above, eg. #c8d7ab. 

So here are some tests with scrub background color #c8d7ab and a new scrub pattern in #b0be93 (jsdotgenerator was used with distance 24, relaxed 3 times with relaxation radius 32 metric 2. Rendered without pixel alignment to get a more naturally fuzzy look.

![twmbarlwm-scrub-c8d7ab](https://user-images.githubusercontent.com/42757252/48940745-1a2d3480-ef5c-11e8-9e01-a7fd50d82777.png)

![caldiera-scrub-c8d7ab](https://user-images.githubusercontent.com/42757252/48940748-1d282500-ef5c-11e8-9a1c-c91463146de5.png)

![ribeira-scrub-c8d7ab](https://user-images.githubusercontent.com/42757252/48940766-274a2380-ef5c-11e8-8e0b-aab54c940b7d.png)

With intermittent streams:
![estrada-scrub-c8d7ab](https://user-images.githubusercontent.com/42757252/48940759-244f3300-ef5c-11e8-91d4-2ddf96df2285.png)

![test-scrub-c8d7ab](https://user-images.githubusercontent.com/42757252/48941179-d4716b80-ef5d-11e8-96cd-7b6cba968831.png)

This color is a little darker (3%) than the previous, but the streams are still visible

**Comment by Adamant36** [comment]
@jeisenbe, I like the new pattern. Whats the advantage over the original though? Maybe the same thing could be done for the forest pattern also at some point.

**Comment by jeisenbe** [comment]
This was the command used to generate the new pattern with jsdotpattern:

http://www.imagico.de/map/jsdotpattern.php#x,512,jdp97762;g5,45,64,64;rx,250,2,64,64;rx,250,2,64,64;rx,250,2,64,64;rd,1,0,0,scrub2,1,5,12,0,jdp3379,8ece8f,b5e3b5;

Compare to the wood/forest pattern:

Actually it looks like the old pattern was only semi-random; this new pattern is more random, and perhaps the icons are a few pixels closer together on average.

I'm thinking of submitting a PR for just this change, and leaving some more time to work on heath and grass later, because this change looks good and resolves this issue.

**Comment by Adamant36** [comment]
I was thinking the new scrub pattern was random the old forest pattern. It might be worth randomizing the forest pattern more at some point. 

As far as the PR goes, you should do it for just this change. Heath is a different issue anyway. Your going to do it with scrub as %3 darker then heath (#d1e0b4) right?

**Comment by jeisenbe** [comment]
Area in Hawaii with streams, intermittent streams on scrub near golf course:

Scrub background color #c8d7ab with scrub pattern in #b0be93:
z15
![new-scrub-z15-hawaii](https://user-images.githubusercontent.com/42757252/48965980-92066800-f00b-11e8-80ee-38c0428ad10c.png)
z16
![new-scrub-hawaii-z16](https://user-images.githubusercontent.com/42757252/48965983-96cb1c00-f00b-11e8-8d47-4f13f52f990c.png)

**Comment by jeisenbe** [comment]
The two landcovers should get their own color in the code, to make it
easier to maintain and adapt in the future.

This issue is about making intermittent waterways more visible on scrub,
with a better-looking scrub rendering as a secondary goal, so I think it
can go first, unless we can all agree on heath soon.

**Comment by Adamant36** [comment]
I'm not exactly sure what the pros and cons of using a variable to represent a color is, but I know they are used a lot in the landcover.mss file. So I don't why it would be an issue. Unless its particular to this case for some reason. That being said, if you can figure out the exact hex value for darken(@heath, 3%) then that's fine. 

Also, it then has the disadvantage of being harder to tweak in small amounts if need be. While most people would know what the difference between darken(@heath, 3%) and darken(@heath, 2.5) might be its much harder and takes more work to interpret and tweak a hex value with by using the lch whatever thing or a color slider in a photo editing app (where you might not be able to do it with that fine of detail). Which is probably why the darken thing is used. 

As I said though, as long its the exact same color, it doesn't really matter how its represented in there. Although, thinking about it, I remember a couple of issues where the hex value was used instead of the darken variable and the maintainers said to use the darken variable instead because it shows how the color was obtained. So....I think there's definitely more of a slant toward using the variable when it can be. Plus, there's plenty in the contributing file about how code should be consistent and use already existing standards. Which the variable thing clearly is. Even if you might personally prefer it the other way.

**Comment by Tomasz-W** [comment]
`#d0e3b6`+ 10% vertical dash pattern for heath + `#c8d6ab` for scrub

![scrub1](https://user-images.githubusercontent.com/25656654/48977451-a04f9500-f09a-11e8-8d69-cd888cec69f5.jpg)
![scrub2](https://user-images.githubusercontent.com/25656654/48977452-a04f9500-f09a-11e8-889a-edc7351ade1c.jpg)
![scrub3](https://user-images.githubusercontent.com/25656654/48977453-a04f9500-f09a-11e8-93e1-fc99233ef30a.jpg)
![scrub4](https://user-images.githubusercontent.com/25656654/48977454-a04f9500-f09a-11e8-83ae-e17dcc3f89dd.jpg)
![scrub5](https://user-images.githubusercontent.com/25656654/48977455-a0e82b80-f09a-11e8-9bd4-55eac2cf7e85.jpg)

**Comment by jeisenbe** [comment]
Ok, it looks like we agree that #c8d6ab will work for scrub. I can make a
PR for this change alone, to resolve this issue. The we can work on heath
later.

**Comment by Adamant36** [comment]
Sounds good to me.
